src/librustc_mir/build/expr/category.rs

```diff
@@ -74,20 +72,15 @@ impl Category {
             ExprKind::Borrow { .. } |
             ExprKind::Assign { .. } |
             ExprKind::AssignOp { .. } |
-            ExprKind::InlineAsm { .. } =>
-                Some(Category::Rvalue(RvalueFunc::AsRvalue)),
+            ExprKind::InlineAsm { .. } => Some(Category::Rvalue(RvalueFunc::AsRvalue)),
 
-            ExprKind::Literal { .. } =>
-                Some(Category::Constant),
+            ExprKind::Literal { .. } => Some(Category::Constant),
 
             ExprKind::Loop { .. } |
             ExprKind::Block { .. } |
             ExprKind::Break { .. } |
             ExprKind::Continue { .. } |
-            ExprKind::Return { .. } =>
-                // FIXME(#27840) these probably want their own
-                // category, like "nonterminating"
-                Some(Category::Rvalue(RvalueFunc::Into)),
+            ExprKind::Return { .. } => Some(Category::Rvalue(RvalueFunc::Into)),
```

Comment:
We lost a comment here - it should probably be moved inside a block (I wonder if rustfmt will still kill it then?)